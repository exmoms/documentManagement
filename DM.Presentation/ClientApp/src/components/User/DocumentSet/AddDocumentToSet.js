/**
 * @file Contains AddDocumentToSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Component } from "react";
import DocumentTable from "../DocumentTable";
import SearchDocuments from "../SearchDocuments";
import CREATE_DOC_STYLES from "../Styles";
import { withStyles } from "@material-ui/core/styles";
import { withTranslation } from "react-i18next";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContentText from "@material-ui/core/DialogContentText";

import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";

const styles = CREATE_DOC_STYLES;

class AddDocumentToSet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      columns: [],
    };
    this.updateParentState = this.props.action;
  }

  getDocumentList = (list, columns) => {
    this.setState({
      data: list,
      columns: columns,
    });
  };

  handleClose = () => {
    this.setState(
      {
        data: [],
        columns: [],
      },
      this.updateParentState()
    );
  };

  /**
   * @todo Could we reduce rendering this component, once it renders two times
   * each time we update state
   */

  t = this.props["t"];
  render() {
    const { classes } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.handleClose()}
          aria-labelledby="form-dialog-title"
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="form-dialog-title">
              {this.t("documentset_addDocumentsToSet_title")}
            </DialogTitle>
          </div>
          <DialogContent>
            <div
              style={
                lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
              }
            >
              <DialogContentText>
                {this.t("documentset_addDocumentsToSet_Content")}
              </DialogContentText>
            </div>
            <TableContainer component={Paper}>
              <Table className={classes.table} aria-label="simple table">
                <TableBody>
                  <TableRow key="search-form">
                    <TableCell>
                      <SearchDocuments
                        getDocumentList={this.getDocumentList}
                        action="add-to-documentset"
                        setIdExclude={this.props.set.id}
                      />
                      <DocumentTable
                        data={this.state.data}
                        columns={this.state.columns}
                        set={this.props.set}
                        action="add-to-documentset"
                        handleClose={this.handleClose}
                      />
                    </TableCell>
                  </TableRow>
                </TableBody>
              </Table>
            </TableContainer>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.handleClose()}
              color="primary"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(AddDocumentToSet));
