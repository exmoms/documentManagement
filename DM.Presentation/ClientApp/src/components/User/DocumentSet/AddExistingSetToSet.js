/**
 * @file Contains AddExistingSetToSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Component, Fragment } from "react";
import { fetchData } from "../../../api/FetchData";
import { Redirect } from "react-router";
import { postDataToAPI } from "../../../api/PostData";
import { withStyles } from "@material-ui/core/styles";
import { withTranslation } from "react-i18next";
import Avatar from "@material-ui/core/Avatar";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Dialog from "@material-ui/core/Dialog";
import AddIcon from "@material-ui/icons/Add";
import { blue } from "@material-ui/core/colors";

const styles = {
  avatar: {
    backgroundColor: blue[100],
    color: blue[600],
  },
};

class AddExistingSetToSet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      setId: 0,
      setsList: [],
      redirectToDocumentSet: false,
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };
    this.updateParentState = this.props.action;
  }

  async componentDidMount() {
    // if (this.props.set.id !== this.state.setId && this.props.set.id !== 0) {
    let id = this.props.set.id;
    let url = `/api/DocumentSet/GetAllSetsExcludingSetsOfSet?Parent_documentSet_Id=${id}`;
    await fetchData(url)
      .then((value) => {
        this.setState({ setId: id, setsList: value });
      })
      .catch((e) => console.log(e));
    // }
  }

  componentDidUpdate() {
    if (this.state.redirectToDocumentSet) {
      this.setState(
        {
          redirectToDocumentSet: false,
        },
        this.handleClose()
      );
    }
  }

  addSetToSet = async (childSetId) => {
    let ParentSetId = this.props.set.id;

    let response = await postDataToAPI(
      `/api/DocumentSet/AddDocumentSetToDocumentSet?Parent_documentSet_Id=${ParentSetId}&child_documentSet_Id=${childSetId}`
    );

    if (response.ok) {
      this.setState(
        {
          redirectToDocumentSet: true,
          openNotifiction: true,
        } //,
        // this.handleClose()
      );
    } else {
      let json = await response.json();
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
          error: true,
          errorMessage: json.error,
        } //,
        //this.handleClose()
      );
    }
  };

  handleClose = () => {
    this.setState(
      {
        setId: 0,
        setsList: [],
        redirectToDocumentSet: false,
      },
      () => {
        this.updateParentState();
      }
    );
  };

  t = this.props["t"];

  render() {
    const { classes, show } = this.props;
    const { setsList } = this.state;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <Fragment>
        <Dialog
          onClose={this.handleClose}
          aria-labelledby="simple-dialog-title"
          open={show}
          maxWidth={"xs"}
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="simple-dialog-title">
              {this.t("documentset_addDocumentSetToSet_title")}
            </DialogTitle>
          </div>
          {setsList.length > 0 ? (
            <List>
              {setsList.map((set, index) => {
                return (
                  <ListItem
                    key={index}
                    button
                    onClick={() => this.addSetToSet(set.id)}
                  >
                    <ListItemAvatar>
                      <Avatar className={classes.avatar}>
                        <AddIcon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary={set.name} />
                  </ListItem>
                );
              })}
            </List>
          ) : (
            <List>
              <ListItem button onClick={() => this.handleClose()}>
                <ListItemAvatar>
                  <Avatar className={classes.avatar}>
                    <AddIcon />
                  </Avatar>
                </ListItemAvatar>
                <ListItemText primary={this.t("documentset_noSetFound")} />
              </ListItem>
            </List>
          )}
        </Dialog>
        {this.state.redirectToDocumentSet && (
          <Redirect
            to={{
              pathname: "/document-set",
              state: {
                id: this.props.set.id,
                name: this.props.set.name,
                openNotifiction: this.state.openNotifiction,
                error: this.state.error,
                errorMessage: this.state.errorMessage,
              },
            }}
          />
        )}
      </Fragment>
    );
  }
}

export default withStyles(styles)(withTranslation()(AddExistingSetToSet));
