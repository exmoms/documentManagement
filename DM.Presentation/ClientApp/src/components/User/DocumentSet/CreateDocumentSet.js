/**
 * @file Contains CreateDocumentSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment } from "react";
import { Redirect } from "react-router";
import { TextField } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import { postDataToAPI } from "../../../api/PostData";
import { withTranslation } from "react-i18next";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";

const styles = (theme) => ({
  root: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  box: {
    padding: "10px",
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 300,
  },
  Button: {
    marginTop: "70px",
  },
});

class CreateDocumentSet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      redirectToDocumentSet: false,
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };

    this.handleClose = this.props.action;
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidUpdate() {
    if (this.state.redirectToDocumentSet) {
      this.setState(
        {
          redirectToDocumentSet: false,
        },
        this.handleClose()
      );
    }
  }

  handleChange(event) {
    this.setState({ name: event.target.value });
  }

  handleSubmit = async (event) => {
    let set = this.props.set;
    let response = {};
    if (set.id === 0) {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
      });
    } else {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
        ParentDocumentSet: set.id,
      });
    }
    if (response.ok) {
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
        } //,
        //this.handleClose()
      );
    } else {
      let json = await response.json();
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
          error: true,
          errorMessage: json.error,
        } //,
        // this.handleClose()
      );
    }
  };

  t = this.props["t"];
  render() {
    const { classes } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <Fragment>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.handleClose()}
          maxWidth={"xs"}
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="simple-dialog-title">
              {this.t("documentset_addnewdocumentset")}
            </DialogTitle>
          </div>
          <DialogContent>
            <div className={classes.root}>
              <div className={classes.box}>
                <div
                  style={
                    lang === "ar"
                      ? { textAlign: "right" }
                      : { textAlign: "left" }
                  }
                >
                  {this.t("documentset_entername")}
                </div>
                <TextField
                  id="standard-basic"
                  className={classes.textField}
                  required
                  helperText={this.t("hypertext_required")}
                  label={this.t("namelabel")}
                  fullWidth
                  value={this.state.name}
                  onChange={this.handleChange}
                />
                <br />
                <Button
                  variant="contained"
                  className={classes.Button}
                  color="primary"
                  fullWidth
                  onClick={this.handleSubmit}
                >
                  {" "}
                  {this.t("button_submit")}
                </Button>
              </div>
            </div>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.handleClose()}
              color="primary"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
        {this.state.redirectToDocumentSet && (
          <Redirect
            to={{
              pathname: "/document-set",
              state: {
                id: this.props.set.id,
                name: this.props.set.name,
                openNotifiction: this.state.openNotifiction,
                error: this.state.error,
                errorMessage: this.state.errorMessage,
              },
            }}
          />
        )}
      </Fragment>
    );
  }
}

export default withStyles(styles, { withTheme: true })(
  withTranslation()(CreateDocumentSet)
);
