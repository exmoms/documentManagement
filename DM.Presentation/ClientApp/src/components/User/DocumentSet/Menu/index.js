import React, { Component } from "react";

import styled from "styled-components";

export default class Menu extends Component {
  render() {
    return (
      <Container>
        {this.props.content.map((c) => (
          <InnerContainer
            key={c.info}
            className="content"
            style={c.style}
            onClick={c.onClick}
          >
            {c.info}
          </InnerContainer>
        ))}
      </Container>
    );
  }
}

const Container = styled.div`
  position: absolute;
  background: white;
  width: 200px;
  z-index: 1000;
  border: 1px solid rgba(221, 224, 228, 0.5);
  box-shadow: 0 12px 24px 0 rgba(0, 0, 0, 0.05);
  border-radius: 8px;
  & .content {
    padding: 15px;
    transition: background 250ms ease-in;
    &:hover {
      background: #eeeff1;
    }
  }
`;

const InnerContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
`;
