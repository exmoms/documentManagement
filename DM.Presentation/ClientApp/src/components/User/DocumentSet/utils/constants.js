const FILE = 0;
const FOLDER = 1;

export { FILE, FOLDER };
