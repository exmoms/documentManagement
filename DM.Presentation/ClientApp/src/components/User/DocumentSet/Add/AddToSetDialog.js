/**
 * @file Contains AddToSetDialog component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Dialog from "@material-ui/core/Dialog";
import AddIcon from "@material-ui/icons/Add";
import { blue } from "@material-ui/core/colors";
import { FOLDER, FILE } from "../utils/constants";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles({
  avatar: {
    backgroundColor: blue[100],
    color: blue[600],
  },
});

export default function AddToSetDialog(props) {
  const classes = useStyles();
  const { show, set } = props;

  const handleClose = () => {
    props.action();
  };

  const { t } = useTranslation();
  let lang = localStorage.getItem("i18nextLng");
  return (
    <Dialog
      onClose={handleClose}
      aria-labelledby="simple-dialog-title"
      open={show}
      maxWidth={"xs"}
    >
      <div
        style={lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }}
      >
        <DialogTitle id="simple-dialog-title">{t("actions")}</DialogTitle>
      </div>
      <List>
        <ListItem autoFocus button onClick={() => props.action(FOLDER, true)}>
          <ListItemAvatar>
            <Avatar className={classes.avatar}>
              <AddIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText primary={t("documentset_addnewdocumentset")} />
        </ListItem>

        {set.id !== 0 ? (
          <Fragment>
            <ListItem
              autoFocus
              button
              onClick={() => props.action(FOLDER, false)}
            >
              <ListItemAvatar>
                <Avatar className={classes.avatar}>
                  <AddIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={t("add_existing_set")} />
            </ListItem>
            <ListItem autoFocus button onClick={() => props.action(FILE)}>
              <ListItemAvatar>
                <Avatar className={classes.avatar}>
                  <AddIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={t("add_document_to_set")} />
            </ListItem>
          </Fragment>
        ) : (
          ""
        )}
      </List>
    </Dialog>
  );
}
