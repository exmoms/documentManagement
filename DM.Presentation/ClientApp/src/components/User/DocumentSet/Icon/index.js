import React, { Component, createRef } from "react";
import Menu from "../Menu";
import UpdateOrPreviewDocumentDialog from "../../UpdateOrPreviewDocumentDialog";
import PrintDocument from "../../PrintDocument";
import ConfirmOperationDialog from "../../ConfirmOperationDialog";
import { postDataToAPI } from "../../../../api/PostData";
import { FILE, FOLDER } from "../utils/constants";
import FileIcon from "../assets/img/file.png";
import FolderIcon from "../assets/img/folder.png";
import { withTranslation } from "react-i18next";
import { Container, Logo, Img, Name } from "./styles";

class Icon extends Component {
  nodeRef = createRef();
  t = this.props["t"];

  state = {
    visible: false,
    showPreviewDocumentDialog: false,
    documentScans: [],
    showPrintDocumentDialog: false,
    showConfirmOperationDialog: false,
    confirmDialogContent: {},
    entryIdToDelete: -1,
  };

  _handleContextMenu = (event) => {
    event.preventDefault();

    const path = event.composedPath();

    const wasOutside = !path.includes(this.nodeRef.current) || false;

    if (wasOutside) {
      this.setState({
        visible: false,
      });
      return;
    }

    this.setState({
      visible: true,
    });
  };

  _handleMouseLeave = (event) => {
    const { visible } = this.state;
    const wasOutside = !(event.target.contains === this.nodeRef.current);

    if (wasOutside && visible)
      this.setState({
        visible: false,
      });
  };

  componentDidMount() {
    document.addEventListener("contextmenu", this._handleContextMenu);

    document.addEventListener("click", this._handleMouseLeave);
  }

  UNSAFE_componentWillUnmount() {
    document.removeEventListener("contextmenu", this._handleContextMenu);

    document.removeEventListener("click", this._handleMouseLeave);
  }

  _handleClick = (event) => {
    const { visible } = this.state;
    const wasOutside = !(event.target.contains === this.nodeRef);

    if (wasOutside && visible)
      this.setState({
        visible: false,
      });
  };

  _handleScroll = () => {
    const { visible } = this.state;

    if (visible)
      this.setState({
        visible: false,
      });
  };

  buildMenuContent = (type, entry) => {
    let { set } = this.props;
    let content = [];
    if (type === FILE) {
      content.push({
        info: this.t("dialog_previewdocument"),
        onClick: () => {
          this.showPreviewDocumentDialog();
        },
      });
      content.push({
        info: this.t("print"),
        onClick: () => {
          this.showPrintDocumentDialog(entry);
        },
      });

      if (set.id !== 0) {
        content.push({
          info: this.t("deleteFromHere"),
          style: { color: "red" },
          onClick: () => {
            this.showConfirmOperationDialog(entry, "delete-document");
          },
        });
      }
    } else {
      content.push({
        info: this.t("open"),
        onClick: () => {
          this.enterFolder(entry.id, entry.name);
        },
      });
      content.push({
        info: this.t("deletePermanently"),
        style: { color: "red" },
        onClick: () => {
          this.showConfirmOperationDialog(entry, "delete-set-permanent");
        },
      });
      if (set.id !== 0) {
        content.push({
          info: this.t("deleteFromHere"),
          style: { color: "red" },
          onClick: () => {
            this.showConfirmOperationDialog(entry, "delete-set-here");
          },
        });
      }
    }
    return content;
  };

  enterFolder = (id, name) => {
    this.props.enterFolder(id, name);
  };

  showConfirmOperationDialog = (entry, option) => {
    let contentText = null;
    switch (option) {
      case "delete-document":
        contentText = this.t("dialog_documentSet_confirmDeleteDocumentFromSet");
        break;
      case "delete-set-here":
        contentText = this.t(
          "dialog_documentSet_confirmDeleteDocumentSetFromDocumentSet"
        );
        break;
      case "delete-set-permanent":
        contentText = this.t(
          "dialog_documentSet_confirmDeleteDocumentSetPermanently"
        );
        break;
      default:
        contentText = "Are you sure to Delete";
    }

    this.setState({
      showConfirmOperationDialog: true,
      entryIdToDelete: entry.id,
      confirmDialogContent: {
        type: option,
        title: this.t("dialog_deleteConfirmation"),
        contentText: contentText,
        buttonText: this.t("delete"),
      },
    });
  };

  handleCloseConfirmOperationDialog = async (ok, option) => {
    if (ok) {
      switch (option) {
        case "delete-document":
          await this.deleteDocumentFormSet(this.state.entryIdToDelete);
          break;
        case "delete-set-here":
          await this.deleteSetFormSet(this.state.entryIdToDelete);
          break;
        case "delete-set-permanent":
          await this.deleteSetPermanently(this.state.entryIdToDelete);
          break;
        default:
      }
    } else {
      this.setState({
        showConfirmOperationDialog: false,
        entryIdToDelete: -1,
        confirmDialogContent: {},
      });
    }
  };

  deleteSetFormSet = async (id) => {
    let { set } = this.props;

    let response = await postDataToAPI(
      `/api/DocumentSet/RemoveDocumentSetFromDocumentSet?Parent_documentSet_Id=${set.id}&child_documentSet_Id=${id}`,
      null,
      "DELETE"
    );

    if (response.ok) {
      this.setState(
        {
          showConfirmOperationDialog: false,
        },
        () => this.props.refreshSet(set)
      );
    } else {
      let json = await response.json();
      this.props.showNotification(true, true, json.error);
    }
  };

  deleteDocumentFormSet = async (id) => {
    let { set } = this.props;
    let response = await postDataToAPI(
      `/api/DocumentSet/RemoveDocumentFromDocumentSet?document_Id=${id}&document_set_Id=${set.id}`,
      null,
      "DELETE"
    );
    if (response.ok) {
      this.setState(
        {
          showConfirmOperationDialog: false,
        },
        () => this.props.refreshSet(set)
      );
    } else {
      let json = await response.json();
      this.props.showNotification(true, true, json.error);
    }
  };

  deleteSetPermanently = async (id) => {
    let { set } = this.props;
    let response = await postDataToAPI(
      `/api/DocumentSet/DeleteDocumentSet?document_set_id=${id}`,
      null,
      "DELETE"
    );
    if (response.ok) {
      this.setState(
        {
          showConfirmOperationDialog: false,
        },
        () => this.props.refreshSet(set)
      );
    } else {
      let json = await response.json();
      this.props.showNotification(true, true, json.error);
    }
  };

  showPreviewDocumentDialog = () => {
    this.setState({ showPreviewDocumentDialog: true });
  };

  ClosePreviewDocumentDialog = () => {
    this.setState({ showPreviewDocumentDialog: false });
  };

  fetchDocumentsScans = async (latestVersion) => {
    // Send list of LatestVersions to api and receive list of scans
    let payload = [];
    payload.push(latestVersion);
    let response = await postDataToAPI(
      "api/Document/GetDocumentScansByVersionIds",
      payload
    );
    return response;
  };

  showPrintDocumentDialog = async (entry) => {
    let response = await this.fetchDocumentsScans(entry.latestVersion);
    if (response.ok) {
      let scans = await response.json();
      let list = [];
      scans.forEach((scan, index) => {
        scan.imgs.forEach((url, idx) => {
          list.push(url);
        });
      });
      this.setState({
        showPrintDocumentDialog: true,
        documentScans: list,
      });
    } else {
      let json = await response.json();
      this.props.showNotification(true, true, json.error);
    }
  };

  ClosePrintDocumentDialog = () => {
    this.setState({
      documentScans: [],
      showPrintDocumentDialog: false,
    });
  };

  render() {
    const { entry, type } = this.props;
    return (
      <Container ref={this.nodeRef}>
        <Logo
          onDoubleClick={() => {
            if (type === FOLDER) {
              this.enterFolder(entry.id, entry.name);
            } else {
              this.showPreviewDocumentDialog();
            }
          }}
        >
          <Img src={type === FOLDER ? FolderIcon : FileIcon} />
        </Logo>
        <Name>{type === FOLDER ? entry.name : entry.documentName}</Name>
        {this.state.visible && (
          <Menu content={this.buildMenuContent(type, entry)} />
        )}

        {this.state.showPreviewDocumentDialog && (
          <UpdateOrPreviewDocumentDialog
            open={this.state.showPreviewDocumentDialog}
            option={0}
            documentVersionId={entry.latestVersion}
            handler={this.ClosePreviewDocumentDialog}
          />
        )}
        {this.state.showPrintDocumentDialog && (
          <PrintDocument
            open={this.state.showPrintDocumentDialog}
            scannedDoc={this.state.documentScans}
            handleClose={this.ClosePrintDocumentDialog}
          />
        )}
        {this.state.showConfirmOperationDialog && (
          <ConfirmOperationDialog
            open={this.state.showConfirmOperationDialog}
            content={this.state.confirmDialogContent}
            handleClose={this.handleCloseConfirmOperationDialog}
          />
        )}
      </Container>
    );
  }
}

export default withTranslation()(Icon);
