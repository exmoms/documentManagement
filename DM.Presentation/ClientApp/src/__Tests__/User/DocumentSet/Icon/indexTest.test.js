import React from "react";
import ReactDom from "react-dom";
import Icon from "../../../../components/User/DocumentSet/Icon/index.js";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import { Fragment } from "react";
import ListItem from "@material-ui/core/ListItem";
import { FOLDER, FILE } from "../../../../components/User/DocumentSet/utils/constants";
import AddToSetDialog from "../../../../components/User/DocumentSet/Add/AddToSetDialog";
import AddDocumentToSet from "../../../../components/User/DocumentSet/AddDocumentToSet";
import AddExistingSetToSet from "../../../../components/User/DocumentSet/AddExistingSetToSet";
import CreateDocumentSet from "../../../../components/User/DocumentSet/CreateDocumentSet";
import { Container, Logo, Img, Name } from "../../../../components/User/DocumentSet/Icon/styles";
import FileIcon from "../../../../components/User/DocumentSet/assets/img/file.png";
import FolderIcon from "../../../../components/User/DocumentSet/assets/img/folder.png";
import UpdateOrPreviewDocumentDialog from "../../../../components/User/UpdateOrPreviewDocumentDialog";
import PrintDocument from "../../../../components/User/PrintDocument";
import Menu from "../../../../components/User/DocumentSet/Menu";
afterEach(cleanup);

describe("Index", () => {
    test("renders without crashing", () => {
        let set={id:1};
        const enterFolder = jest.fn();
        const refreshSet = jest.fn();
        const showNotification = jest.fn();
        let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
        let type=FOLDER;
        const div = document.createElement("div");
        ReactDom.render(<Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/>, div);
      });

    test("Icon matches snapshot", () => {
        let set={id:1};
        const enterFolder = jest.fn();
        const refreshSet = jest.fn();
        const showNotification = jest.fn();
        let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
        let type=FOLDER;
    const renderedValue = createMount()(
        <Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/>
      );
      expect(renderedValue.html()).toMatchSnapshot();
    });
   
    test("intializations", () => {
        let set={id:1};
        const enterFolder = jest.fn();
        const refreshSet = jest.fn();
        const showNotification = jest.fn();
        let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
        let type=FOLDER;
        const wrapper = mount(
            <Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/>
          );
          expect(wrapper.find(Img).at(0).props().src).toBe(FolderIcon);
          expect(wrapper.find(Name).text()).toBe("name A");
          expect(wrapper.find(Menu).length).toBe(0);
          expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(0);
          expect(wrapper.find(PrintDocument).length).toBe(0);
          wrapper.setProps({type:FILE});
          wrapper.update();
          wrapper.render();
          expect(wrapper.find(Img).at(0).props().src).toBe(FileIcon);
          expect(wrapper.find(Name).text()).toBe("documentName A");
          expect(wrapper.find(Menu).length).toBe(0);
          expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(0);
          expect(wrapper.find(PrintDocument).length).toBe(0);

      });
       //TBD
      test("componentDidMount", () => {
        let set={id:1};
        const enterFolder = jest.fn();
        const refreshSet = jest.fn();
        const showNotification = jest.fn();
        let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
        let type=FOLDER;
        const map = {};
           window.addEventListener = jest.fn((event, cb) => {
          map[event] = cb;
          });
        const addEventListenerMock = jest.spyOn(document, 'addEventListener');
          const preventDefaultMock = jest.fn();
          const composedPathMock = jest.fn(()=>{return []});
          var event = new Event("contextmenu", { preventDefault: preventDefaultMock, composedPath:composedPathMock ,clientX:1, clientY:1});
        const wrapper = mount(
            <Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/> , { attachTo: document.body });
         // map.contextmenu({ preventDefault: preventDefaultMock, composedPath:composedPathMock ,clientX:1, clientY:1})
         document.dispatchEvent(event);
         expect(addEventListenerMock).toHaveBeenCalledTimes(2);
          expect(addEventListenerMock.mock.calls[0][0]).toBe('contextmenu');
          expect(addEventListenerMock.mock.calls[1][0]).toBe('click');
        });
      
        test("onDoubleClick", () => {
          let set={id:1};
          const enterFolder = jest.fn();
          const refreshSet = jest.fn();
          const showNotification = jest.fn();
          let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
          let type=FOLDER;
          const wrapper = mount(
              <Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/>
            );
            expect(wrapper.find(Img).at(0).props().src).toBe(FolderIcon);
            expect(wrapper.find(Name).text()).toBe("name A");
            expect(wrapper.find(Menu).length).toBe(0);
            expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(0);
            expect(wrapper.find(PrintDocument).length).toBe(0);
            act(() => {
              wrapper
                .find(Logo)
                .at(0)
                .props()
                .onDoubleClick();
            });
            wrapper.update();
            wrapper.render();
            expect(enterFolder).toHaveBeenCalledTimes(1);
            expect(enterFolder.mock.calls[0][0]).toBe(1);
            expect(enterFolder.mock.calls[0][1]).toBe("name A");
        });

        test("onDoubleClick when type is FILE and handler", () => {
          let set={id:1};
          const enterFolder = jest.fn();
          const refreshSet = jest.fn();
          const showNotification = jest.fn();
          let entry={id:1 , name:"name A" , documentName:"documentName A" ,latestVersion :2};
          let type=FILE;
          const wrapper = mount(
              <Icon set={set} enterFolder={enterFolder} refreshSet={refreshSet} showNotification={showNotification} entry={entry} type={type}/>
            );
            expect(wrapper.find(Img).at(0).props().src).toBe(FileIcon);
            expect(wrapper.find(Name).text()).toBe("documentName A");
            expect(wrapper.find(Menu).length).toBe(0);
            expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(0);
            expect(wrapper.find(PrintDocument).length).toBe(0);
            act(() => {
              wrapper
                .find(Logo)
                .at(0)
                .props()
                .onDoubleClick();
            });
            wrapper.update();
            wrapper.render();
            expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(1);
            expect(wrapper.find(UpdateOrPreviewDocumentDialog).at(0).props().open).toBeTruthy();

            act(() => {
              wrapper
                .find(UpdateOrPreviewDocumentDialog)
                .at(0)
                .props()
                .handler();
            });
            wrapper.update();
            wrapper.render();
            expect(wrapper.find(UpdateOrPreviewDocumentDialog).length).toBe(0);
        });
});