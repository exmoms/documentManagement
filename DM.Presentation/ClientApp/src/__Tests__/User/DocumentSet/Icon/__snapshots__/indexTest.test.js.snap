// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Icon matches snapshot 1`] = `"<div class=\\"sc-AxheI kEOPNl\\"><div class=\\"sc-Axmtr ilxBVU\\"><img src=\\"folder.png\\" class=\\"sc-AxmLO cAKCgh\\"></div><div class=\\"sc-fzozJi jxgYyc\\">name A</div></div>"`;
